### ISSUE TITLE
Suggestion: Select browser source as window

### ISSUE THREAD

**Issue by Nizars** [LorenzCK/OnTopReplica#52]
I know that this request isn't easy to implement but I will suggest it regardless due to the huge demand for it.

Context:
Streamers on Twitch use OBS to stream games but also have a need to read the Twitch chat to interact with their viewers. There is currently only one paid app that I know of that allows the streamer to read the chat as an overlay on top of the game and it is called PlayClaw.
It is a good program and it get's the job done but it is quite heavy since it is a full program for streaming.

Having the program accept a browser source link similar to the example bellow would be very beneficial to a lot of people:
![image](https://user-images.githubusercontent.com/16970186/40851360-7ed9e73a-65c7-11e8-9565-5b296e7f59fd.png)

I understand that this is going to require a lot of work and expansion but I am suggesting it in case there is an interest from the developers.

**Comment by blundercode** [comment]
You can do this, Just select browser as your source then use Region select on the window to only grab the chat part of the window. Then just don't minimize that browser leave it running in the back behind everything else like always.

**Comment by Nizars** [comment]
Hi Erik,

The point of using a browser source rather than simply using a currently opened browser as a source is to not have a browser open in the background. 

When you are streaming, your machine is doing all it can to both deliver a satisfactory gaming performance and encode the game/window capture content at the desired resolution, refresh rate and bit rate. 

You don’t want to have any other applications taking from the available resources. 

It kind of defeats the purpose to have a browser running in the background and this app running on top of your video game with it running in borderless windowed mode. 

It would be smarter to just open the chat in a browser directly on another monitor and run the game in full screen mode. 

The goal is to allow the streamer to read the chat without having a web browser running somewhere in the background. 

I hope it makes sense.

**Comment by Nizars** [comment]
OBS for example allows users to use a browser source as a source, just like you would use a window source in OnTopReplica. 

It however only renders the added sources on the stream which is understandable because it does not come with any window/application overlay to render such a scene on. 

You can read more about it here:
https://github.com/obsproject/obs-studio/wiki/Sources-Guide#browsersource

**Comment by Nizars** [comment]
What I would love to see is a solution that combines both, OBS’s ability to add a browser source (not a window source), and OnTopReplica’s ability to render a source on a screen. 

I believe that such solution would has the potential to find a lot of popularity within the streaming community for reasons too numerous to go into. There simply isn’t any such solution available, especially a free or a light weight one. 

I am willing to learn how the browser source implementation is done in OBS and attempt to introduce it to this project as a fork/branch if there is support from the devs here.

**Comment by LorenzCK** [comment]
Absolutely! I am terribly behind with OnTopReplica and appreciate any kind of contribution.
I just checked and apparently there is a [CEF NuGet package for WinForms](https://www.nuget.org/packages/CefSharp.WinForms/). If this works, this could be the easiest way to embed a browser into OTR.

**Comment by Nizars** [comment]
Music to my ears

**Comment by blundercode** [comment]
I've streamed using OnTopReplica as well it's not very resource intensive at all, might get you by until you can figure something else out. It's basically the equivalent of hovering over a taskbar item, quite lightweight. Would be cool to do what you are saying though.

[https://github.com/LorenzCK/OnTopReplica/issues/26](url) You might find this one interesting as well.

**Comment by ryantheleach** [comment]
Embedding browsers is notoriously annoying, watching the OBS dev chatter.

I'd use the existing apps that can do this instead, or rely upon your streaming applications to do it.

As for the performance considerations of running a browser in the background, if you even have 1 other tab open, it is likely more performant to capture from the browser, then using CEF as CEF will create an entirely new chrome process embedded in an application.

This issue is invalid IMO, for multiple reasons.

That said, it would be a cool feature, just not if the author believes it is feature-creep, or as a 'performant' alternative to running browsers.

In addition, embedded browsers have difficulties with cookies and authentication, as (for better or worse) cookies and sessions are not stored between CEF and browsers.
